docs/compiler/compiler.rst

```diff
@@ -0,0 +1,325 @@
+    
+    Compiles a ``Callable`` or set of them into an Ivy graph. If ``args`` or ``kwargs`` are specified, 
+    compilation is performed eagerly, otherwise, compilation will happen lazily.
+    
+    :param objs: Callable(s) to compile and create a graph of.
+    :type objs: ``Callable``
+    :param stateful: List of instances to be considered stateful during the graph compilation.
+    :type stateful: ``Optional[List]``
+    :param arg_stateful_idxs: Positional arguments to be considered stateful during the graph compilation.
+    :type arg_stateful_idxs: ``Optional[List]``
+    :param kwarg_stateful_idxs: Keyword arguments to be considered stateful during the graph compilation.
+    :type kwarg_stateful_idxs: ``Optional[List]``
+    :param to: Backend that the graph will be compiled to. If not specified, the current backend will be used.
+    :type to: ``str``
+    :param include_generators: Include array creation/generation functions as part of the graph.
+    :type include_generators: ``bool``
+    :param array_caching: Cache the constant arrays that appear as arguments to the functions in the graph.
+    :type array_caching: ``bool``
+    :param return_backend_compiled_fn: Whether to apply the native compilers, i.e. tf.function, after ivy's compilation.
+    :type return_backend_compiled_fn: ``bool``
+    :param static_argnums: For jax's jit compilation.
+    :type static_argnums: ``Optional[Union[int, Iterable[int]]]``
+    :param static_argnames: For jax's jit compilation.
+    :type static_argnames: ``Optional[Union[str, Iterable[str]]]``
+    :param args: Positional arguments for obj.
+    :type args: ``Optional[Tuple]``
+    :param kwargs: Keyword arguments for obj.
+    :type kwargs: ``Optional[dict]``
+    :rtype: ``Union[Graph, LazyGraph, ivy.Module, ModuleType]``
+    :return: A compiled ``Graph`` or a non-initialized ``LazyGraph``. If the object is an ``ivy.Module``, the forward pass will be compiled and the same module will be returned. If the object is a ``ModuleType``, the function will return a copy of the module with every method lazily compiled.
+
+Using the compiler
+------------------
+
+To use the ``ivy.compile()`` function, you need to pass a callable object and the corresponding inputs
+to the function.
+
+Let's start with a simple function:
+
+.. code-block:: python
+
+    import ivy
+
+    ivy.set_backend("torch")
+
+    def fn(x, y):
+        z = x**y
+        print(z)
+        k = x * y
+        j = ivy.concat([x, z, y])
+        sum_j = ivy.sum(j)
+        return z
+
+    x = ivy.array([1, 2, 3])
+    y = ivy.array([2, 3, 4])
+
+    # Compile the function
+    compiled_fn = ivy.compile(fn, args=(x, y))
+
+    # View the graph
+    compiled_fn.show()
+
+In this case, the compiled graph would be:
+
+.. image:: https://raw.githubusercontent.com/unifyai/unifyai.github.io/master/img/externally_linked/compiler/figure1.png
```

Comment:
This image seems too big, is there a way to standardize the size based on the nodes? I guess probably the easiest thing is to resize the image with extra background

Reply:
fixed this